kernel/src/scan/mod.rs

```diff
@@ -119,10 +119,14 @@ impl ScanBuilder {
 /// A vector of this type is returned from calling [`Scan::execute`]. Each [`ScanResult`] contains
 /// the raw [`EngineData`] as read by the engines [`crate::ParquetHandler`], and a boolean
 /// mask. Rows can be dropped from a scan due to deletion vectors, so we communicate back both
-/// EngineData and information regarding whether a row should be included or not See the docs below
-/// for [`ScanResult::mask`] for details on the mask.
+/// EngineData and information regarding whether a row should be included or not (via an internal
+/// mask). See the docs below for [`ScanResult::full_mask`] for details on the mask.
 pub struct ScanResult {
-    /// Raw engine data as read from the disk for a particular file included in the query
+    /// Raw engine data as read from the disk for a particular file included in the query. Note
+    /// that this data may include data that should be filtered out based on the mask given by
+    /// [`full_mask`].
+    ///
+    /// [`full_mask`]: #method.full_mask
```

Comment:
Never seen `#method.full_mask` Is there a name/example of this kind of doc? I would've expected something like `ScanResult::full_mask`.

Reply:
cool! ty :D